dls-controls/malcolmjs
# Feature/widget array #269

## Description

Display method parameters which have type malcolm:core/*ArrayMeta:1.0 as tables with a single column (+1 for alarm state). 

## Testing instructions

Add a set up instructions describing how the reviewer should test the code
- Review code
- Check Travis build
- Review changes to test coverage
- edit public/settings.json to connect to a malcolm instance running DEMO-SCANNING.yaml (or another configuration with a scan configure/validate method)
- npm start
- navigate to localhost:3000/SCAN/validate.takes.axesToMove
- see the array parameter view, add a new row and type in "x" or "y"
- enter the following generator into the json editor for that param (under the validate method): {"typeid": "scanpointgenerator:generator/CompoundGenerator:1.0", "excluders": [], "continuous": true, "generators": [{"typeid": "scanpointgenerator:generator/LineGenerator:1.0", "alternate": true, "axes": ["x"], "stop": [1.0], "start": [0.0], "units": ["mm"], "size": 5}, {"typeid": "scanpointgenerator:generator/LineGenerator:1.0", "alternate": false, "axes": ["y"], "stop": [1.0], "start": [0.0], "units": ["mm"], "size": 6}], "duration": 0.5, "mutators": []}
- run the validate method
- click on the button to view the axesToMove from the results.
- view the read-only/return array param

## Agile board tracking

connect to #269 
closes #269
